src/utils.ts

```diff
@@ -10,6 +11,26 @@ export function parseNumber(n: string): number {
 	return num;
 }
 
+// Note: this could be a good place to put any additional call parsing logic or
+// even argument parsing logic.
+// Takes in a call or an array that contains a call and recursively parses all calls.
+export function parseCalls(data: any): any {
```

Comment:
It may be better to put this function directly inside ApiHandler, since that is probably the only place it will ever be used for. In general, "utility" files are sort of considered an anti-pattern and should be used _very_ carefully. Ref http://ralin.io/blog/oop-anti-patterns-utility-or-helper-classes.html